autoload/go/cmd.vim

```diff
@@ -124,6 +135,35 @@ function! go#cmd#Test(compile, ...)
     endif
 endfunction
 
+function! go#cmd#TestFocused(...)
+    let test = search("func Test", "bcsW")
+
+    if test == 0
+        echo "vim-go: [test] no test found immediate to cursor"
+    else
+        let line = getline(test)
+        let name = split(split(line, " ")[1], "(")[0]
+        let flag = "-run '" . name . "$'"
+
+        let a1 = ""
+        if len(a:000)
+            let a1 = a:1
+
+            " add extra space
+            let flag = " " . flag 
+        endif
+
+        call go#cmd#Test(0, a1, flag)
+    end
+endfunction
+
+" default cache, in case TestLast is called before anything
+let s:go_cmd_test_args_cache = [0, ""]
+
+function! go#cmd#TestLast()
```

Comment:
My main though for this was, 

1. `:GoTestFunction`
2. Fix failing code (most likely in another buffer, tab, etc..)
3. `:GoTestLast` (To rerun the test, without jumping back to the test buffer)

However, having this on my own setup for the last week or so, I haven't found myself going back to this too often. Might just be muscle memory of going back to the test or just running `:GoTest`, not sure...

Now that I think about, not terribly good with names, but something like `:GoTestReplay` or `:GoTestRerun` might sound better?

I'll remove it from this PR and open another for this specifically to marinate over it a bit longer.